### ISSUE TITLE
使用pytorch和tensorrt结果相差很大

### ISSUE THREAD

**Issue by momohuangsha** [mit-han-lab/temporal-shift-module#181]
I trained  2 classes of behavior recognition using MobilenetV2 and then used Torch and TensorRT on the Jetson Nano to get vastly different results. Excuse me, what's the problem?

model = TSN(
    num_class=2, 
    num_segments=8, 
    modality="RGB",
    base_model="mobilenetv2", 
    dropout=0.8,
    is_shift=True, 
    before_softmax=True)
state_dict = torch.load("checkpoint/epoch_59_ckpt.pth.tar")['state_dict']
model.load_state_dict(state_dict)
model.cuda()
model.eval()

x = torch.randn((24, 224, 224)).cuda()
model_trt = torch2trt_dynamic(model, [x])
......
out = model(buffer)   # buffer shape [24,224,224]
out_trt = model_trt(buffer)

results：
out          tensor([[ 1.6837, -1.6845]], device='cuda:0')
out_trt     tensor([[-4.6641,  4.7363]], device='cuda:0')
====================================
out           tensor([[ 1.6734, -1.6750]], device='cuda:0')
out_trt      tensor([[-4.6915,  4.7649]], device='cuda:0')

tensorrt (7.1.3.0)
topi (0.6.0)
torch (1.6.0)
torch2trt (0.2.0)
torch2trt-dynamic (0.3.0)

**Comment by momohuangsha** [comment]
I found the cause of the problem:

out[:, :-1, :fold] = x[:, 1:, :fold]  # shift left
out[:, 1:, fold: 2 * fold] = x[:, :-1, fold: 2 * fold]  # shift right
out[:, :, 2 * fold:] = x[:, :, 2 * fold:]  # not shift

in  tensorrt  out  will be  a constant 
and also when  use torch.onnx.export to save  as onnx model ,there will be a warning below:
TracerWarning: Converting a tensor to a Python index might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  out[:, :-1, :fold] = x[:, 1:, :fold]  # shift left

TracerWarning: Converting a tensor to a Python index might cause the trace to be incorrect. We can't record the data flow of Python values, so this value will be treated as a constant in the future. This means that the trace might not generalize to other inputs!
  out[:, 1:, fold: 2 * fold] = x[:, :-1, fold: 2 * fold]  # shift right

so,is there any other way to replace the above three lines of code

**Comment by irvingzhang0512** [comment]
Maybe you can try another implementation of shift module

https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/backbones/resnet_tsm.py#L40-L121

**Comment by momohuangsha** [comment]
> Maybe you can try another implementation of shift module
> 
> https://github.com/open-mmlab/mmaction2/blob/master/mmaction/models/backbones/resnet_tsm.py#L40-L121

@irvingzhang0512   Thanks !  It works now
